test/kubetest/k8s.go

```diff
@@ -1230,31 +1230,73 @@ func (k8s *K8s) CreateTestNamespace(namespace string) (string, error) {
 }
 
 // CreateServiceAccounts create service accounts with passed names
-func (k8s *K8s) CreateServiceAccounts() ([]*v1.ServiceAccount, error) {
-	rv := make([]*v1.ServiceAccount, 0, len(k8s.sa))
-	for _, n := range k8s.sa {
-		sa, err := k8s.clientset.CoreV1().ServiceAccounts(k8s.namespace).Create(&v1.ServiceAccount{
-			ObjectMeta: metaV1.ObjectMeta{
-				Name: n,
-			},
+func (k8s *K8s) CreateServiceAccounts() {
+	accountCount := len(k8s.sa) + 1 //1 means default acc
+	errs := make(chan error, accountCount)
+	for i := range k8s.sa {
+		index := i
+		go func() {
+			n := k8s.sa[index]
+			sa, err := k8s.clientset.CoreV1().ServiceAccounts(k8s.namespace).Create(&v1.ServiceAccount{
+				ObjectMeta: metaV1.ObjectMeta{
+					Name: n,
+				},
+			})
+			if err != nil {
+				errs <- err
+				return
+			}
+			err = waitFor(accountWaitTimeout, func() bool {
+				sa, getErr := k8s.clientset.CoreV1().ServiceAccounts(k8s.namespace).Get(n, metaV1.GetOptions{})
+				if getErr != nil {
+					logrus.Errorf("An error during get service account: %v, err: %v", sa.Name, getErr.Error())
+					return false
+				}
+				logrus.Info(sa)
+				return len(sa.Secrets) != 0
+			})
+			if err != nil {
+				err = errors.Wrapf(err, "account: %v, secrets == 0", sa.Name)
+			}
+			errs <- err
+		}()
+	}
+
+	go func() {
+		err := waitFor(accountWaitTimeout, func() bool {
+			_, getErr := k8s.clientset.CoreV1().ServiceAccounts(k8s.namespace).Get(pods.DefaultAccount, metaV1.GetOptions{})
```

Comment:
Is it possible to extract predicate for ServiceAccount checking to separate function, that parametrised with ServiceAccount's name?